### ISSUE TITLE
It is not possible to keep manually added properties and update existing ones.

### ISSUE THREAD

**Issue by ca-stefan-cordes** [mulesoft/mule-maven-plugin#462]
The configuration of mule-maven-plugin allows to set <overrideProperties>false</overrideProperties> 
which is fine for keeping manually added properties like 
```
secure.properties.key=..........
```
when re-deploying a new application.
see /mule-deployer/src/main/java/org/mule/tools/deployment/cloudhub/CloudHubArtifactDeployer.java => resolveProperties(..)

But when setting overrideProperties to false **all** properties which already exist in the worker are kept.
Due to that e.g. changing						`<org.mule.runtime.core.api.processor.strategy.AsyncProcessingStrategyFactory.DEFAULT_MAX_CONCURRENCY>16</org.mule.runtime.core.api.processor.strategy.AsyncProcessingStrategyFactory.DEFAULT_MAX_CONCURRENCY>`
to
`<org.mule.runtime.core.api.processor.strategy.AsyncProcessingStrategyFactory.DEFAULT_MAX_CONCURRENCY>32</org.mule.runtime.core.api.processor.strategy.AsyncProcessingStrategyFactory.DEFAULT_MAX_CONCURRENCY>`
in the pom.xml does not have any effect because the original value 16 is taken in resolveProperties(...)

So we need to deploy with overrideProperties=true to get the updated MAX_CONCURRENCY but then worker is not starting
because secure.properties.key is missing. It needs to be added manually again (which makes automatic deployment useless).

As there are several solutions to solve this I first want to start discussion before creating a pull request.

Solution 1: Just keep manually **added**
- take entries of originalApplication.getProperties() for those ones not in deployment.getProperties()

Solution 2: Specify property names to be **kept**
- Add a List to configuration instead of overrideProperties:
`<keepPropertyNames><keepPropertyName></keepPropertyName></keepPropertyNames>`
where all properties are specyfied which sould be taken from originalApplication.getProperties().
In our case: `<keepPropertyName>secure.properties.key</keepPropertyName>`

Solution 3: Specify property names to **override always**
- Add a List to configuration instead of overrideProperties:
`<forceOverwritePropertyNames><forceOverwritePropertyName></forceOverwritePropertyName></forceOverwritePropertyNames>`
In our case: `<forceOverwritePropertyName>org.mule.runtime.core.api.processor.strategy.AsyncProcessingStrategyFactory.DEFAULT_MAX_CONCURRENCY</forceOverwritePropertyName>`

**Comment by ca-stefan-cordes** [comment]
Any comments?

**Comment by ca-stefan-cordes** [comment]
You may vote for this idea: 
"Add upsert option for properties in mule-maven-plugin"
https://help.mulesoft.com/s/ideas#08734000000HLUfAAO

**Comment by ca-stefan-cordes** [comment]
Any idea whether properties in cloudhub2 are handled better?

**Comment by thebluenova** [comment]
@ca-stefan-cordes 
Running in to this, or something very similar attempting to deploy from Azure DevOps using maven to CloudHub 2.0.
out of interest where are you setting <overrideProperties>true</overrideProperties> I'm struggling to find any mention of it in the MuleSoft documentation. I'm guessing it goes somewhere in pom.xml?
I've found it in the Mule-maven source code, but it looks like only in CloudHub 1?

**Comment by ca-stefan-cordes** [comment]
Yes, the boolean is placed inside the cloudHubDeployment section in the pom.xml: (Cloudhub1):

```
						<groupId>org.mule.tools.maven</groupId>
						<artifactId>mule-maven-plugin</artifactId>
						<configuration>
							<cloudHubDeployment>
							     ....
							     <overrideProperties>true</overrideProperties>
```

Yes, it is for Cloudhub1 only.
For cloudhub2 ( Cloudhub2Deployment extends RuntimeFabricDeployment ) 
currently no "overrideProperties" or "mergeProperties" is foreseen in mule-deployer. Just secureProperties can be specified:
org.mule.tools.model.anypoint.RuntimeFabricDeployment.secureProperties

As `org.mule.tools.deployment.cloudhub2.RequestBuilderCh2.createConfiguration()` is just placing properties and secureProperties into the request, I guess the server part will replace everything by the provided properties (so overrideproperties is true by default for Cloudhub2) => so no merge/keep there, too.

**Comment by ca-stefan-cordes** [comment]
Any news?

**Comment by ca-stefan-cordes** [comment]
Ping

**Comment by quaddy-services** [comment]
With https://github.com/mulesoft/mule-maven-plugin/pull/833 it will be possible to initially set some properties and later keep properties in pom.xml empty to maintain (and keep) properties via Runtime Manager only.

**Comment by SimoneTrovatoFCG** [comment]
Hello, in the actual `main` branch (corresponding to 4.4.0 release) I don't see this feature merged:

![Image](https://github.com/user-attachments/assets/d5bd3baa-b628-4eeb-bdbf-1e50aba69ea6)

Am I wrong?

**Comment by SimoneTrovatoFCG** [comment]
The code included in [PR #833](https://github.com/mulesoft/mule-maven-plugin/pull/833) seems to be released with 4.2.0 version:

![Image](https://github.com/user-attachments/assets/830859a7-505d-47d8-9952-099de48ef83c)

...but starting from 4.2.1 has been replaced by the code previous to 4.2.0

**Comment by SimoneTrovatoFCG** [comment]
Any news?

**Comment by ssisaias** [comment]
This is so much missed when we are using automated deployments...

**Comment by ca-stefan-cordes** [comment]
As this repo is dead you need to contact mulesoft on a different channel.
See https://github.com/mulesoft/mule-maven-plugin/pull/1009
